### PR TITLE
Add Default Sharing Token

### PR DESCRIPTION
**Pull request by rezafaghani**
Hi bro
I add a endpoint to share payment between IBN account that register in Asan Pardakht .
I added it from documented Asan Pardakht share with me and of course test it with real parameter and its worked.

### CODE REVIEW

File: ApIpgSample/Controllers/HomeController.cs [hunk 1]
```diff
@@ -76,7 +76,7 @@ public async Task<IActionResult> Pay(PayViewModel data)
         {
```
Comment (mrdaghestani) [inline ApIpgSample/Controllers/HomeController.cs:76]: This is a public repository, so it's better to give an option to users for choosing token generation with or without default sharing.
I suggest you add a button in the buy form with the "Pay with Default Sharing" title so when I click it, use `GenerateBuyDefaultSharingToken` unless use `GenerateBuyToken`.
Thanks

File: AsanPardakht.IPG/Abstractions/IServices.cs [hunk 2]
```diff
@@ -21,6 +21,18 @@ public interface IServices
...
+      /// first you should register IBN in Asan Pardakht and then 
+      /// Asan pardakht do it automatically
+      /// for more info you should talk with AP 
+      /// </summary>
+      /// <param name="amountInRials"></param>
+      /// <param name="callbackURL"></param>
+      /// <param name="paymentId"></param>
+      /// <param name="mobileNumber"></param>
+      /// <returns></returns>
+        Task<GenerateTokenResponse> GenerateBuyDefaultSharingToken(ulong amountInRials, string callbackURL, string paymentId = null, string mobileNumber = null);
```
Comment (mrdaghestani) [inline AsanPardakht.IPG/Abstractions/IServices.cs:35]: It's better to name the method like this:

```suggestion
        Task<GenerateTokenResponse> GenerateBuyTokenWithDefaultSharing(ulong amountInRials, string callbackURL, string paymentId = null, string mobileNumber = null);
```

File: AsanPardakht.IPG/Services.cs [hunk 3]
```diff
@@ -60,7 +60,34 @@ public async Task<GenerateTokenResponse> GenerateToken(GenerateTokenRequest data
...
 
         }
+        /// <summary>
+        /// add default sharing
+        /// </summary>
+        /// <param name="data"></param>
+        /// <returns></returns>
+        /// <exception cref="InvalidIdentityException"></exception>
+        /// <exception cref="AsanPardakhtIPGException"></exception>
+        public async Task<GenerateTokenResponse> GenerateShareToken(GenerateTokenRequest data)
```
Comment (mrdaghestani) [inline AsanPardakht.IPG/Services.cs:70]: Isn't it better to name it as follows?

```suggestion
        public async Task<GenerateTokenResponse> GenerateTokenWithDefaultSharing(GenerateTokenRequest data)
```

**Review by mrdaghestani — CHANGES_REQUESTED** [review]
Thank you for your contribution.
Please check and fix the issues that I mentioned after that I'll approve your PR so the NuGet packages will update automatically.
Thanks

File: ApIpgSample/Controllers/HomeController.cs (hunk 1 above)
Comment (rezafaghani) [reply ApIpgSample/Controllers/HomeController.cs:76]: Cause there is two different way to share amount and this name is in official document i think that it's better to not change the name to suggested by you.
By save the official name user can find quickly and understand the method.

File: AsanPardakht.IPG/Abstractions/IServices.cs (hunk 2 above)
Comment (rezafaghani) [reply AsanPardakht.IPG/Abstractions/IServices.cs:35]: well its better to not change the method name different from official document. so users can find it with a sample search and so user can match the name with the official document name

File: AsanPardakht.IPG/Services.cs (hunk 3 above)
Comment (rezafaghani) [reply AsanPardakht.IPG/Services.cs:70]: well as same as the others . do not change the name to different name from official name. and yes i think your name is better

**Comment by mrdaghestani** [comment]
Hey, I made it in another way.
Try it and let me know is it fine for your use case?
Thanks

**Pull request closed without merging**